### ISSUE TITLE
Find and open folders through omnibox

### ISSUE THREAD

**Issue by MartinLichtblau** [rpkamp/chrome-minimal-bookmarks-tree#6]
Hello :)
would it be possible to search and open also for bookmarks-folders through the omnibox? Opening each bookmark in the folder one by one, makes the speed improvement of the omnibox access meaningless. 
There is [only one extension](https://chrome.google.com/webstore/detail/context-bookmarks/ednnkpljcbpjcoplcooifiblghkephlh) which can do that, but it's basic functions are not as good as yours and it's the dev is inactive. 
I appreciate your great work, but this feature would be great!
_With best greetings
M_L_

**Comment by rpkamp** [comment]
I will look into this, but it's not trivial and does not have major priority for me. Just know it's on the radar.

**Comment by MartinLichtblau** [comment]
If it's so hard to do, you might like to "take a look" at this extension which already has this functionality implemented. It only has a bug, that it doesn't open some folders, beginning with special letters.

**Comment by rpkamp** [comment]
Basically the problem I'm having is that the only thing you can use in the Omnibox API is text. That is, if I have two bookmarks that have the same title in two different folders and I pick one of them, the extension will just get the title of the bookmark as input and doesn't know which one of the two bookmarks with that name was meant. Now this in itself is annoying enough but doesn't happen a lot so you most likely won't run in to it. Note that the other extension you mentioned has the same problem and doesn't solve it either.

Now when you put folders in the mix this problem gets even worse because instead of picking the one bookmark you meant it might just open a folder with a lot of bookmarks and open a lot of tabs and make chrome unresponsive for a while. Bad for the user, bad for the reviews of my extension, bad all around.

One thing I might do (thinking out loud here) is add an extra keyword indicating you're looking to open a folder. So suppose you're looking to open a folder called 'CNET' you would type `bm folder CNET` or something like that. Also I would make an option for it that is disabled by default to prevent people from accidentally opening a lot of tabs without them meaning to.

Doesn't that sound workable to you? I'm open to other suggestions as well if you have any!

**Comment by MartinLichtblau** [comment]
Thanks for the comprehensive explanation of the underlying issues!  

To prevent opening a folder by mistake, you could show a folder icon, and for sites a star.
And if it's still an issue, an optional dialog saying "Do you really want to open XX bookmarks at once" should stop really everyone from opening to many sites at once by mistake.

**Comment by rpkamp** [comment]
This feature is coming to Minimal Bookmarks Tree v2.0, which is currently in the making. I'll let you know when it's there :smile:

**Comment by rpkamp** [comment]
I've just pushed version 2.0.0 to the chrome web store, so it should arrive in your browser shortly :)

If you can't wait, go to [chrome extensions][1], tick "development mode" and click "update extensions now".

[1]: chrome://extensions/

**Comment by MartinLichtblau** [comment]
Yeah, I got it! 
It works smooth and recognizes if bookmarks have been changed. Just two little points from an usability perspective which I would change. First, it open them, and leaves the current empty tab empty, and opens the bookmarks behind it, so you have to manually close this one. Normally when you type and open an URL it opens in the same tab. Second, while you type it shows bookmark/folder matching your search, but when you click enter nothing happens, so you have to manually select the one. Just do it like it works in every browser, open the first suggested item when the user hits enter. So, doing so it would work like the users expects it, since everywhere it behaves like this. 
In short: 1. Open in current tab 2. Open first entry when hitting enter

With best greetings from Germany, have a great sunny day wherever you are! :-)

**Comment by rpkamp** [comment]
Thanks for your feedback! I will have a look and see if I can improve this. 

Since you're from Germany, could I ask you if you would be interested in translating this extension to German? I have a site I could point you to where you could enter translations easily. It's only 42 sentences, most of which are just a few words.

**Comment by MartinLichtblau** [comment]
Yes, I would like to do that! (y)

**Comment by rpkamp** [comment]
Cool!

The website with the translations is https://minimalbookmarkstree.oneskyapp.com/translate/sign/in. You need to create an account there if you don't already have one, and supply me your username so I can add you as a collaborator. Once I've done that you can edit the translations.

**Comment by MartinLichtblau** [comment]
My username there is *MartinLichtblau*

**Comment by rpkamp** [comment]
I see you've translated all the texts, thank you!
I will see if I can upload a new version with the German strings tonight :)

**Comment by MartinLichtblau** [comment]
That's lighting fast! 
FYI.: I updated some of my translations.

**Comment by rpkamp** [comment]
I've just published the latest version (v2.2.0) with your translations in it!

I made some minor adjustments to the translations, but left most as is.

Thanks again! :)

**Comment by rpkamp** [comment]
> In short: 1. Open in current tab 2. Open first entry when hitting enter

This has been fixed in the latest version that is rolling out now (2.2.2)

**Comment by MartinLichtblau** [comment]
Ahh, I see what you did there ;-) 
Thanks for including my two suggestions!

**Comment by MartinLichtblau** [comment]
Is there a way to set an easier omnibox keyword?
For b+m you really have to think and doesn't feel great. Since we have our index finger sitting on F, and F is  often used to search, I would like to use it as the keyword. Quick to access and fluent to continue writing the search term.

**Comment by rpkamp** [comment]
No, there is not. It used to be configurable by the user in older chrome versions, but for some reason it's not anymore. If chrome would have an API to change this dynamically so I could expose this as a setting I would, but it doesn't, so I'm afraid we're stuck with bm.
If it helps, I picked bm because these are the first characters of the two syllables in **b**ook**m**ark :bookmark: 

Also, the extension has way too many users for me to be changing the keyword at this point for all of them, so I won't do that.
